tests/servicebus_functions/servicebus_trigger_return/__init__.py

```diff
@@ -0,0 +1,5 @@
+import azure.functions as azf
+
+
+def main(msg: azf.QueueMessage) -> bytes:
```

Comment:
Not sure if I am looking at the right file. Is this right? QueueMessage? Should it be ServiceBusMessage? It will be easier if you just paste function.json, __init__.py

Reply:
Yea, that's a different trigger, disregard it for now. The correct trigger is in `tests/servicebus_functions/servicebus_trigger/__init__.py`:

```
import json

import azure.functions as azf


def main(msg: azf.ServiceBusMessage) -> str:
    result = json.dumps({
        'message_id': msg.message_id,
        'body': msg.get_body().decode('utf-8'),  # the body here is invalid 
        'content_type': msg.content_type,
        'expiration_time': msg.expiration_time,
        'label': msg.label,
        'partition_key': msg.partition_key,
        'reply_to': msg.reply_to,
        'reply_to_session_id': msg.reply_to_session_id,
        'scheduled_enqueue_time': msg.scheduled_enqueue_time,
        'session_id': msg.session_id,
        'time_to_live': msg.time_to_live,
        'to': msg.to,
        'user_properties': msg.user_properties,
    })

    return result
```

`function.json`:

```json
{
  "scriptFile": "__init__.py",
  "disabled": false,

  "bindings": [
    {
      "type": "serviceBusTrigger",
      "direction": "in",
      "name": "msg",
      "queueName": "testqueue",
      "connection": "AzureWebJobsServiceBusConnectionString",
    },
    {
      "type": "blob",
      "direction": "out",
      "name": "$return",
      "connection": "AzureWebJobsStorage",
      "path": "python-worker-tests/test-servicebus-triggered.txt"
    }
  ]
}

```

Reply:
Thanks. This helps. Test code that inserts item into testqueue?

Reply:
I tried different variations with JavaScript functions - HttpTrigger to ServiceBus Output binding and then verifying serviceBusTrigger get invoked. I cannot repro this issue. 

function.json and test that triggers looks OK.

At this point, we need to debug host. Here are the instructions, please give it a try
- Clone https://github.com/Azure/azure-functions-host
- Open in VSCode
- Set following environment variables
  - **languageWorkers:python:workerDirectory** to  **{Dir}\azure-functions-python-worker**
  - **consoleLoggingMode** to always
  - **AzureWebJobsScriptRoot** to your test directory with host.json
- Set breakpoints at
  - [RpcMessageConversionExtensions.cs#L23](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Rpc/MessageExtensions/RpcMessageConversionExtensions.cs#L23) - Possible issue converting data from languageworker 
 - [RpcMessageConversionExtensions.cs#L48](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Rpc/MessageExtensions/RpcMessageConversionExtensions.cs#L48) - Possible issue converting data when sending to language worker
- F5 and invoke httpTrigger that has the serviceBus output binding